### PR TITLE
[netcore] Fix build.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mcs/class/System.Private.CoreLib/System/Buffer.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public static void BlockCopy (Array src, int srcOffset, Array dst, int dstOffset
 
 		static bool IsPrimitiveTypeArray (Array array)
 		{
-			throw new NotImplementedException ();
+			return array.GetType ().GetElementType ().IsPrimitive;
```
Comment (EgorBo) [inline mcs/class/System.Private.CoreLib/System/Buffer.cs]: I wonder why .net core has to do a pinvoke for it, performance reasons?

Comment (vargaz) [reply mcs/class/System.Private.CoreLib/System/Buffer.cs]: This code needs to be optimized later.

Comment (filipnavara) [reply mcs/class/System.Private.CoreLib/System/Buffer.cs]: @EgorBo Yes, performance reasons, avoiding reflection. Also it's FCall, not P/Invoke, which usually points to something performance sensitive especially when the native code doesn't erect a frame using `HELPER_METHOD_FRAME_BEGIN_*`.

Comment (EgorBo) [reply mcs/class/System.Private.CoreLib/System/Buffer.cs]: maybe we should document somewhere or use a special attribute to mark places where we put a naive implementation to revise it later?

**Pull request merged**